### DISCUSSION TITLE
Accessing "missing" DPs

### DISCUSSION THREAD

**Discussion by hazzabeee** [jasonacox/tinytuya#373] [Q&A] [answered]
Hello. Bit of an odd one. I have a smart pet feeder. OutletDevice.detect_available_dps() returns the following DPs (which I think is a correct representation of DP_QUERY):

`{'dps': {'3': 3, '4': 'standby', '14': 0, '15': 3}}`

However, cloud.getdps() returns the following:

```
{"code": "meal_plan", "dp_id": 1, "type": "Raw", "values": "{}"},
{"code": "manual_feed", "dp_id": 3, "type": "Integer", "values": "{"min":1,"max":12,"scale":0,"step":1}"},
{"code": "feed_state", "dp_id": 4, "type": "Enum", "values": "{"range":["standby","feeding"]}"},
{"code": "factory_reset", "dp_id": 9, "type": "Boolean", "values": "{}"},
{"code": "feed_report", "dp_id": 15, "type": "Integer", "values": "{"min":0,"max":12,"scale":0,"step":1}"}
```

That is, OutletDevice.detect_available_dps() seems to return a smaller set of DPs than the cloud equivalent. Is there any way to get it to show more DPs? In particular, I would like to (locally) query dp_id=1 ("meal_plan"). Is there any way of doing that? Many thanks.

**Comment by uzlonewolf** [comment, marked as answer]
Some DPs are not sent in status requests, they're only sent asynchronously when the value changes.  Try running this:
```python
import time
import tinytuya

d = tinytuya.Device( 'your_device_id_here', persist=True )

print( 'Found:', d )
print( 'Status:',  d.status() )

print(" > Begin Monitor Loop <")
pingtime = time.time() + 9

while(True):
    if( pingtime <= time.time() ):
        payload = d.generate_payload(tinytuya.HEART_BEAT)
        data = d.send(payload)
        pingtime = time.time() + 9
        if data:
            print( time.time(), data )

    data = d.receive()
    if data:
        print( time.time(), data )
```
and while that's running make a change to the schedule in the app.

**Reply by hazzabeee** [reply]
Thanks - it's a shame Tuya doesn't allow you to query specific DPs. In my case meal_plan is only updated every couple of months, so it probably makes sense for me to query it via the cloud API once a day, rather than locally every ten seconds. Really appreciate the answer though -- I'm sure it will be useful to others.
